crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -374,6 +381,117 @@ impl Persistence {
 
         Ok(solution)
     }
+
+    /// Returns the oldest settlement event for which the accociated auction is
+    /// not yet populated in the database.
+    pub async fn get_settlement_without_auction(
+        &self,
+    ) -> Result<Option<domain::eth::Event>, DatabaseError> {
+        let _timer = Metrics::get()
+            .database_queries
+            .with_label_values(&["get_settlement_without_auction"])
+            .start_timer();
+
+        let mut ex = self.postgres.pool.begin().await?;
+        let event = database::settlements::get_settlement_without_auction(&mut ex)
+            .await?
+            .map(|event| {
+                let event = domain::eth::Event {
+                    block: u64::try_from(event.block_number)
+                        .map_err(|_| anyhow::anyhow!("negative block"))?
```

Comment:
nit: we should be able to apply this at a bunch of places which currently have `.map_err(|_| anyhow!(""))`
```suggestion
                        .context("negative block")?
```